java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java

```diff
@@ -721,25 +853,35 @@ private ArrowBuf readIntoBuffer(
         default:
           throw new UnsupportedOperationException("Cannot read array of type " + type);
       }
+    } else if (bufferType.equals(VIEWS)) {
+      vectorTypes.add(VARIADIC_DATA_BUFFERS);
```

Comment:
Don't mutate the argument here. Instead below we can look at typeLayout and add VARIADIC_DATA_BUFFERS